<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chromatin accessibility data • ConDecon</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Chromatin accessibility data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ConDecon</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">API</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_Intro_to_ConDecon.html">Quick set up and example</a></li>
    <li><a class="dropdown-item" href="../articles/a2_B_cell_RNA.html">B-cells transcriptomic data</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Spatial_RNA.html">Spatial transcriptomic data</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Chromatin_Accessibility_Data.html">Chromatin accessibility data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/CamaraLab/ConDecon" aria-label="GitHub"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Chromatin accessibility data</h1>
            
      

      <div class="d-none name"><code>a4_Chromatin_Accessibility_Data.Rmd</code></div>
    </div>

    
    
<p>ConDecon’s general approach for estimating cell abundances can be
applied to other omics data modalities such as chromatin accessibility
data. In this tutorial, we will apply ConDecon to bulk and single-cell
ATAC-seq data from patient-derived melanoma cell cultures where the
SOX10 transcription factor was knocked down by siRNA. Cells were then
sampled for sequencing at 0, 24, 48, and 72 hours after SOX10 knockdown
(<a href="https://doi.org/10.1038/s41592-019-0367-1" class="external-link">Bravo González-Blas
et al, Nature Methods, 2018</a>). For convenience, the processed data
associated with this tutorial is available on <a href="https://zenodo.org/record/7604026#.Y-EzEC_MIuU" class="external-link">Zenodo</a>.</p>
<p>Note: To apply ConDecon to chromatin accessibility data, the bulk and
single-cell ATAC-seq data must contain a common set of peaks. We created
this by binning the genome into non-overlapping 10KB bins and
aggregating the peaks from the single-cell and bulk ATAC-seq data into
those bins.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ConDecon</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
<p>We will use single-cell ATAC-seq data from each of the four sampling
times as a reference dataset.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snATAC_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../snATAC_count.Rds"</span><span class="op">)</span></span>
<span><span class="va">snATAC_SVD</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../snATAC_SVD.Rds"</span><span class="op">)</span></span>
<span><span class="va">snATAC_var_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../snATAC_var_genes.Rds"</span><span class="op">)</span></span>
<span><span class="va">snATAC_meta_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../snATAC_meta_data.Rds"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Visualize the sampling time of the single-cell ATAC-seq data</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">snATAC_meta_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP_1</span>, y <span class="op">=</span> <span class="va">UMAP_2</span>, color <span class="op">=</span> <span class="va">hrs</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_ATAC_single_cell_hrs.png" width="600px"></p>
<p>We are interested in using ConDecon to deconvolve 8 bulk ATAC-seq
samples from 2 different patients (MM087 and MM057) and each of the four
sampling times.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BulkATAC_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../BulkATAC_data.Rds"</span><span class="op">)</span></span>
<span><span class="va">BulkATAC_meta_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"../BulkATAC_meta_data.Rds"</span><span class="op">)</span></span></code></pre></div>
<p>We can now use ‘RunConDecon’ to deconvolve bulk ATAC-seq data using
single-cell ATAC-seq data as reference.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">startTime</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ConDecon_obj</span> <span class="op">=</span> <span class="fu"><a href="../reference/RunConDecon.html">RunConDecon</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">snATAC_count</span>, </span>
<span>                           latent <span class="op">=</span> <span class="va">snATAC_SVD</span><span class="op">[</span>,<span class="fl">2</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span>, </span>
<span>                           bulk <span class="op">=</span> <span class="va">BulkATAC_data</span>, </span>
<span>                           variable.features <span class="op">=</span> <span class="va">snATAC_var_genes</span><span class="op">)</span></span>
<span></span>
<span><span class="va">endTime</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.time</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">endTime</span> <span class="op">-</span> <span class="va">startTime</span><span class="op">)</span></span>
<span><span class="co">#Time difference of 1.620496 mins</span></span></code></pre></div>
<p>With ‘PlotConDecon’, we can visualize the relative cell probabilities
of each bulk sample. We find that ConDecon infers a higher abundance of
reference cells from the same sampling time as the query bulk sample,
independently of the specific patient cell line of the query sample.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's visualize the cell abundance distribution of all 8 bulk samples</span></span>
<span><span class="fu"><a href="../reference/PlotConDecon.html">PlotConDecon</a></span><span class="op">(</span>ConDecon_obj <span class="op">=</span> <span class="va">ConDecon_obj</span>, </span>
<span>             umap <span class="op">=</span> <span class="va">snATAC_meta_data</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP_1"</span>, <span class="st">"UMAP_2"</span><span class="op">)</span><span class="op">]</span>, </span>
<span>             pt.size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_ATAC_PlotConDecon.png" width="1000px"></p>
<p>We can now use the cell probabilities inferred by ConDecon to
deconvolve the sampling time of the bulk ATAC-seq data. Using
‘TransferFeatures’, we will estimate the sampling time of each bulk
sample and compare these to the actual sampling time.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ConDecon_obj</span> <span class="op">=</span> <span class="fu"><a href="../reference/TransferFeatures.html">TransferFeatures</a></span><span class="op">(</span>ConDecon_obj <span class="op">=</span> <span class="va">ConDecon_obj</span>, </span>
<span>                                feature <span class="op">=</span> <span class="va">snATAC_meta_data</span><span class="op">$</span><span class="va">time_hrs</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">meta_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>bulk_time <span class="op">=</span> <span class="va">BulkATAC_meta_data</span><span class="op">$</span><span class="va">hrs</span>, </span>
<span>                        transfer_feature <span class="op">=</span> <span class="va">ConDecon_obj</span><span class="op">$</span><span class="va">TransferFeatures</span><span class="op">[</span><span class="fl">1</span>,<span class="op">]</span>, </span>
<span>                        patients <span class="op">=</span> <span class="va">BulkATAC_meta_data</span><span class="op">$</span><span class="va">patient</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">meta_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">bulk_time</span>, y <span class="op">=</span> <span class="va">transfer_feature</span>, color <span class="op">=</span> <span class="va">patients</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">3</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_ATAC_LinePlot.png" width="700px"></p>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Rachael Aubin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
